browser/caliban.py

```diff
@@ -22,93 +23,70 @@
 from skimage.segmentation import find_boundaries
 
 from imgutils import pngify
+from helpers import is_npz_file, is_trk_file
 from config import AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY
 
 
-class ZStackReview:
+class BaseReview(object):  # pylint: disable=useless-object-inheritance
+    """Base class for all Review objects."""
 
-    def __init__(self, filename, input_bucket, output_bucket, subfolders, rgb=False):
+    def __init__(self, filename, input_bucket, output_bucket, subfolders,
+                 raw_key='raw', annotated_key='annotated'):
         self.filename = filename
         self.input_bucket = input_bucket
         self.output_bucket = output_bucket
-        # subfolders is actually the full file path
-        self.subfolders = subfolders
-        self.rgb = rgb
-        self.trial = self.load(filename)
-        self.raw = self.trial["raw"]
-        self.annotated = self.trial["annotated"]
-        self.feature = 0
-        self.feature_max = self.annotated.shape[-1]
-        self.channel = 0
+        self.subfolders = subfolders  # full file path
 
         self.current_frame = 0
+        self.scale_factor = 1
+        self._y_changed = False
+        self.info_changed = False
 
-        if self.rgb:
-            # possible differences between single channel and rgb displays
-            self.dims = len(self.raw.shape)
-            if self.dims == 3:
-                self.raw = np.expand_dims(self.raw, axis=0)
-                self.annotated = np.expand_dims(self.annotated, axis=0)
-            self.max_frames, self.height, self.width, self.rgb_channels = self.raw.shape
-            self.channel_max = 1
-
-            self.rescale_raw()
-            self.reduce_to_RGB()
+        self.color_map = plt.get_cmap('viridis')
+        self.color_map.set_bad('black')
 
-        else:
-            self.max_frames, self.height, self.width, self.channel_max = self.raw.shape
+        self.feature = 0
+        self.channel = 0
 
-        self.dimensions = (self.width, self.height)
+        self.trial = self.load(filename)
+        self.raw = self.trial[raw_key]
+        self.annotated = self.trial[annotated_key]
 
-        # create a dictionary that has frame information about each cell
-        # analogous to .trk lineage but do not need relationships between cells included
-        self.cell_ids = {}
-        self.cell_info = {}
+        self.channel_max = self.raw.shape[-1]
+        self.feature_max = self.annotated.shape[-1]
+        self.max_frames = self.raw.shape[0]
+        self.height = self.raw.shape[1]
+        self.width = self.raw.shape[2]
 
-        for feature in range(self.feature_max):
-            self.create_cell_info(feature)
+        self._x_changed = False
+        self._y_changed = False
```

Comment:
`self._y_changed` is initialized in line 42, `self._x_changed` should be moved there or `self.info_changed` should be moved here.